### PR TITLE
data: define interface for `read_last_scalars`

### PR DESCRIPTION
**Pull request by yatbear**
This will be used for `hparams.backend_context.read_last_scalars`.

Googlers, see b/292102513 for more context.

Tested internally at cl/577906864.

#hparams

### CODE REVIEW

File: tensorboard/data/provider.py [hunk 1]
```diff
@@ -248,6 +248,42 @@ def read_scalars(
         """
         pass
 
+    @abc.abstractmethod
+    def read_last_scalars(
+        self,
+        ctx=None,
+        *,
+        experiment_id,
+        plugin_name,
```
Comment (arcra) [inline tensorboard/data/provider.py:257]: Since we're requiring plugin_name to be specified, I would think having the method be read_latest_values, or something like this, would be more generic/flexible, and we can say in the doc string that it's possible that not all plugins are supported by all implementations.

Although the return value does get a bit more difficult to specify and deal with, so if you prefer this because of this reason, I'm ok with it.

File: tensorboard/data/provider.py [hunk 2]
```diff
@@ -248,6 +248,42 @@ def read_scalars(
...
+        run_tag_filter=None,
+    ):
+        """Read the most recent values from scalar time series.
+
+        Note that some impementations might not support not all scalar types.
```
Comment (arcra) [inline tensorboard/data/provider.py:262]: Grammar nit: might not support not all? Can you rephrase it? Also, did you mean to say not all "plugin" types?

Comment (yatbear) [reply tensorboard/data/provider.py]: Oops, that's a typo, thanks for catching this! Also I I forgot scalars are all numeric (https://www.tensorflow.org/api_docs/python/tf/summary/scalar), I think I was mixing the concept of scalar with tensor. In this case we don't need this note at all, removed.

File: tensorboard/data/provider.py (hunk 1 above)
Comment (yatbear) [reply tensorboard/data/provider.py:257]: I specified `plugin_name` because `read_scalars` also requires `plugin_name`. I prefer the simpler definition for just the scalars, in this case do you think we can just remove the `plugin_name` arg?

Comment (yatbear) [reply tensorboard/data/provider.py:254]: I decided to remove the `plugin_name` completely. I think it's unlikely that we'll have a new plugin type that contains scalar data in the near future. And since scalar seems to be the most used type, we don't have to generalize it to `read_last_values` just yet, wdyt?

Comment (arcra) [reply tensorboard/data/provider.py:254]: Oh, sorry for the late response. I thought I had replied to this. Technically, the plugin type is somewhat independent from the data type, e.g. there's a `custom_scalars` plugin in our examples, I believe, which also uses scalar values.

For this reason, and to be consistent with the rest of the interface for the data provider, I'd lean a bit towards keeping the plugin_name... but this is ok too. I agree it's unlikely we'll ever need anything other than "scalars", and if we do, we can change it later (but again, unlikely).

**Review by arcra — APPROVED** [review]

File: tensorboard/data/provider.py (hunk 1 above)
Comment (yatbear) [reply tensorboard/data/provider.py:257]: Great point, added it back.

Comment (yatbear) [reply tensorboard/data/provider.py:257]: Also I didn't realize I also needed to implement this method in `event_processing/data_provider`. PTAL, thanks!

File: tensorboard/data/grpc_provider_test.py [hunk 3]
```diff
@@ -230,6 +230,60 @@ def test_read_scalars(self):
...
+            "run2": {
+                "tag2": provider.ScalarDatum(
+                    step=10000, wall_time=1235.0, value=0.50
+                ),
+            },
+        }
+
+        self.assertEqual(actual, expected)
+
+        req = data_provider_pb2.ReadScalarsRequest()
```
Comment (arcra) [inline tensorboard/data/grpc_provider_test.py:278]: Consider using named parameters to build protos, like we do with some protos above. Something like:

```
expected_req = data_provider_pb2.ReadScalarsRequest(
    experiment_id = "123",
    plugin_filter = some_proto.PluginFilter(plugin_name = "scalars"),
    ...
)
```

**Review by arcra — APPROVED** [review]

File: tensorboard/data/grpc_provider_test.py (hunk 3 above)
Comment (yatbear) [reply tensorboard/data/grpc_provider_test.py]: Done.

**Review by arcra — APPROVED** [review]
